### PR TITLE
[release-v1.11] Cherry-pick E2E tests for channel: TLS key pair rotation #3406

### PR DESCRIPTION
**Pull request by Leo6Leo**
Cherry-pick https://github.com/knative-extensions/eventing-kafka-broker/pull/3406/files

### CODE REVIEW

**Review by creydr — APPROVED** [review]
/lgtm

File: control-plane/pkg/reconciler/channel/channel.go [hunk 1]
```diff
@@ -346,9 +346,8 @@ func (r *Reconciler) reconcileKind(ctx context.Context, channel *messagingv1beta
 		addressableStatus.Addresses = []duckv1.Addressable{httpAddress}
 	}
 
-	address := addressableStatus.Address.URL.URL()
-	proberAddressable := prober.Addressable{
-		Address: address,
+	proberAddressable := prober.ProberAddressable{
+		AddressStatus: &addressableStatus,
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/channel/channel.go:350]: Somehow, we missed the changes at line 322 and 340 from the original commit https://github.com/knative-extensions/eventing-kafka-broker/commit/828b81ede7af938cdffd01f6f103992469330728

Comment (Leo6Leo) [reply control-plane/pkg/reconciler/channel/channel.go:350]: interesting, thanks for pointing this out!! I will make a PR to fix it.

Comment (pierDipi) [reply control-plane/pkg/reconciler/channel/channel.go:350]: Already fixed!

**Pull request merged**